README.md
# Modeling of costs and risks of active monitoring

[![DOI](https://zenodo.org/badge/48653754.svg)](https://zenodo.org/badge/latestdoi/48653754)

This repository houses code and data for the preprint manuscript "Reducing the threat of imported infectious disease outbreaks with active monitoring" by Nicholas Reich, Justin Lessler, Jay Varma, and Neil Vora. The manuscript is currently under review.

The repository is in the form of an R package, which can be installed using the following commands. You only need to run the first line if you don't already have the `devtools` package installed.
```
install.packages('devtools')
devtools::install_github('reichlab/activemonitr')
```

## Manuscript
Code to run the analyses for the manuscript can be found in: 
[activemonitr/inst/analysis-code/](https://github.com/reichlab/activemonitr/tree/master/inst/analysis-code)

The manuscript itself is dynamically generated, reproducible, and can be found in: 
[activemonitr/inst/manuscript/](https://github.com/reichlab/activemonitr/blob/master/inst/manuscript/active-monitoring.Rmd)

## Web app
There is [an open-source web app](https://www.iddynamics.jhsph.edu/apps/connect/activemonitr/) that accompanies the manuscript. The code for the Shiny app that accompanies the publication can be found in:
[activemonitr/inst/shiny/](https://github.com/reichlab/activemonitr/tree/master/inst/shiny)

## Data
Raw data used for the analyses can be found in: 
[activemonitr/inst/raw-data/](https://github.com/reichlab/activemonitr/tree/master/inst/raw-data)
The data used has been aggregated from several publications and a public data repository, including

 - Virlogeux, V., Park, M., Wu, J. T. & Cowling, B. J. Association between Severity of MERS-CoV Infection and Incubation Period. Emerging Infectious Diseases 22, (2016).
 - Virlogeux V, Park M, Wu JTK, Cowling BJ (2015) Data from: Association between severity of MERS-CoV infection and incubation period. Dryad Digital Repository. [http://dx.doi.org/10.5061/dryad.v3546]
 - Nishiura, H. Determination of the appropriate quarantine period following smallpox exposure: an objective approach using the incubation period distribution. Int J Hyg Environ Health 212, 97–104 (2009).
 - Nishiura, H. & Eichner, M. Infectiousness of smallpox relative to disease age: estimates based on transmission network and incubation period. Epidemiol. Infect. 135, 1145–1150 (2007).
 - Faye, O. et al. Chains of transmission and control of Ebola virus disease in Conakry, Guinea, in 2014: an observational study. The Lancet Infectious Diseases 15, 320–326 (2015).



ChangeLog
ChangeLog file for activeMonitr

v0.0.2 - (2020-02-05) Updated plotting function to include smoother integration over posterior distribution and unknown exposure time, with contributions from Stephen Lauer.

v0.0.1 - Initial release of package



R/data-ebola_gamma_pstr.R
#' Posterior distribution of Ebola incubation period
#'
#' A dataset containing samples from an estimated posterior distribution of the
#' incubation period of Ebola. Distribution was estimated from de-identified
#' data from Faye et al. (2015).
#'
#' @format A data frame with 3,000,000 rows and 5 variables:
#' \itemize{
#'   \item \code{shape} shape parameter of gamma distribution
#'   \item \code{scale} scale parameter of gamma distribution
#'   \item \code{idx} index of sample
#'   \item \code{median} the median correponding to this posterior sample
#'   \item \code{p95} the 95th percentile correponding to this posterior sample
#' }
#' @docType data
#' @references Faye, O. et al. Chains of transmission and control of Ebola virus disease in Conakry, Guinea, in 2014: an observational study. The Lancet Infectious Diseases 15, 320–326 (2015).
#' @name pstr_gamma_params_ebola
#' @usage data(pstr_gamma_params_ebola)
NULL


#'
#' Posterior distribution of MERS-CoV incubation period
#'
#' A dataset containing samples from an estimated posterior distribution of the
#' incubation period of MERS-CoV. Distribution was estimated from de-identified
#' data from Virlogeux et al. (2016).
#'
#' @format A data frame with 1,000,000 rows and 6 variables:
#' \itemize{
#'   \item \code{shape} shape parameter of gamma distribution
#'   \item \code{scale} scale parameter of gamma distribution
#'   \item \code{idx} index of sample
#'   \item \code{median} the median correponding to this posterior sample
#'   \item \code{p95} the 95th percentile correponding to this posterior sample
#'   \item \code{chain} which chain the sample belongs to
#'   }
#' @docType data
#' @references Virlogeux, V., Park, M., Wu, J. T. & Cowling, B. J. Association between Severity of MERS-CoV Infection and Incubation Period. Emerging Infectious Diseases 22, (2016).
#' @name pstr_gamma_params_mers
#' @usage data(pstr_gamma_params_mers)
NULL


#'
#' Posterior distribution of smallpox incubation period
#'
#' A dataset containing samples from an estimated posterior distribution of the
#' incubation period of smallpox. Distribution was estimated from de-identified
#' data from several sources (see References).
#'
#' @format A data frame with 1,000,000 rows and 6 variables:
#' \itemize{
#'   \item \code{shape} shape parameter of gamma distribution
#'   \item \code{scale} scale parameter of gamma distribution
#'   \item \code{idx} index of sample
#'   \item \code{median} the median correponding to this posterior sample
#'   \item \code{p95} the 95th percentile correponding to this posterior sample
#'   \item \code{chain} which chain the sample belongs to
#'   }
#' @docType data
#' @references
#' \itemize{
#'   \item Litvinjenko, S., Arsic, B. & Borjanovic, S. Epidemiologic Aspects of Smallpox in Yugoslavia in 1972. Bulletin of the World Health Organization (1972).
#'   \item Mack, T. M. Smallpox in Europe, 1950-1971. Journal of Infectious Diseases 125, 161–169 (1972).
#'   \item Nishiura, H. Determination of the appropriate quarantine period following smallpox exposure: an objective approach using the incubation period distribution. Int J Hyg Environ Health 212, 97–104 (2009).
#' }
#' @name pstr_gamma_params_smallpox
#' @usage data(pstr_gamma_params_smallpox)
NULL


#'
#' Bootstrapped parameter estimates of 2019-nCoV incubation period
#'
#' A dataset containing samples from an estimated bootstrap distribution of the
#' incubation period of 2019-nCoV was estimated from data
#' collected and stored at https://github.com/HopkinsIDD/ncov_incubation.
#'
#' @format A data frame with 1,000 rows and 2 variables:
#' \itemize{
#'   \item \code{meanlog} meanlog parameter of log-normal distribution
#'   \item \code{sdlog} sdlog parameter of log-normal distribution
#'   \item \code{idx} index of sample
#'   \item \code{median} the median correponding to this bootstrapped sample
#'   \item \code{p95} the 95th percentile correponding to this bootstrapped sample
#'   }
#' @docType data
#' @references
#' \itemize{
#'   \item Lauer, S. A., Grantz, K. H., Bi, Q., Jones, F. K., Zheng, Q., Meredith, H., Azman, A. S., Reich, N. G., & Lessler, J. The incubation period of 2019-nCoV from publicly reported confirmed cases: estimation and application. medRxiv (2020).
#' }
#' @name boot_lnorm_params_ncov
#' @usage data(boot_lnorm_params_ncov)
NULL


#'
#' Kernel Density Estimates for posterior distributions
#'
#' @format Output from fit_kde() function in inst/analysis-code/inc-per-mcmc.R. Data used in plotting credible regions of posterior estimates.
#' @docType data
#' @name kde_smallpox
#' @name kde_ebola
#' @name kde_mers
#' @name kde_ncov
#' @usage data(kde_ebola)
NULL
